components/script/dom/audiotracklist.rs

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::audiotrack::AudioTrack;
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::AudioTrackListBinding::{self, AudioTrackListMethods};
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::reflect_dom_object;
+use crate::dom::bindings::root::{Dom, DomRoot};
+use crate::dom::bindings::str::DOMString;
+use crate::dom::eventtarget::EventTarget;
+use crate::dom::window::Window;
+use dom_struct::dom_struct;
+
+#[dom_struct]
+pub struct AudioTrackList {
+    eventtarget: EventTarget,
+    tracks: DomRefCell<Vec<Dom<AudioTrack>>>,
+}
+
+impl AudioTrackList {
+    pub fn new_inherited(tracks: &[&AudioTrack]) -> AudioTrackList {
+        AudioTrackList {
+            eventtarget: EventTarget::new_inherited(),
+            tracks: DomRefCell::new(tracks.iter().map(|track| Dom::from_ref(&**track)).collect()),
+        }
+    }
+
+    pub fn new(window: &Window, tracks: &[&AudioTrack]) -> DomRoot<AudioTrackList> {
+        reflect_dom_object(
+            Box::new(AudioTrackList::new_inherited(tracks)),
+            window,
+            AudioTrackListBinding::Wrap,
+        )
+    }
+
+    pub fn len(&self) -> usize {
+        self.tracks.borrow().len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.tracks.borrow().is_empty().clone()
+    }
+
+    pub fn item(&self, idx: usize) -> Option<DomRoot<AudioTrack>> {
+        self.tracks
+            .borrow()
+            .get(idx)
+            .map(|track| DomRoot::from_ref(&**track))
+    }
+
+    pub fn enabled_index(&self) -> Option<usize> {
+        self.tracks
+            .borrow()
+            .iter()
+            .position(|track| track.enabled())
+    }
+
+    pub fn set_enabled(&self, idx: usize, value: bool) {
```

Comment:
I filed https://github.com/servo/servo/issues/22799 to make this interact with the media backend. Could you add a TODO comment linking to that issue here and in the equivalent video code?